media-proxy/src/mesh/conn_rdma.cc

```diff
@@ -0,0 +1,388 @@
+#include <queue>
+#include "conn_rdma.h"
+
+namespace mesh {
+
+namespace connection {
+
+Rdma::Rdma() : ep_ctx(nullptr), init(false), trx_sz(0), mDevHandle(nullptr)
+{
+    std::memset(&ep_cfg, 0, sizeof(ep_cfg));
+}
+
+Rdma::~Rdma() { shutdown_rdma(context::Background()); }
+
+std::string Rdma::kind_to_string(Kind kind)
+{
+    switch (kind) {
+    case Kind::undefined:
+        return "undefined";
+    case Kind::transmitter:
+        return "transmitter";
+    case Kind::receiver:
+        return "receiver";
+    default:
+        return "unknown";
+    }
+}
+
+void Rdma::notify_cq_event()
+{
+    std::lock_guard<std::mutex> lock(cq_mutex);
+    event_ready = true;
+    cq_cv.notify_one();
+}
+
+void Rdma::init_buf_available() { buf_available = false; }
+
+void Rdma::notify_buf_available()
+{
+    buf_available.store(true, std::memory_order_release);
+    buf_available.notify_one();
+}
+
+void Rdma::wait_buf_available()
+{
+    buf_available.wait(false, std::memory_order_acquire);
+    buf_available = false;
+}
+
+// Add to queue
+Result Rdma::add_to_queue(void *element)
+{
+    if (!element) {
+        return Result::error_bad_argument;
```

Comment:
Change where applicable.
```suggestion
        return set_result(Result::error_bad_argument);
```
This method automatically increments the error metric automatically on error results.

Reply:
fixed in https://github.com/OpenVisualCloud/Media-Communications-Mesh/pull/286/commits/1e7f66891946f5a904408fb568ca119b7e04eaa0